src/player.cpp

```diff
@@ -4657,8 +4657,8 @@ void player::update_stamina( int turns )
             stamina_recovery += bonus;
         }
     }
-
-    stamina = roll_remainder( stamina + stamina_recovery * turns );
+    //  recover ratio was balanced for legacy 6s/turn, hence the division by 6
+    stamina = roll_remainder( stamina + stamina_recovery * turns / 6.0f );
```

Comment:
That int in there makes me nervous about truncation errors, I'm confident this version will not truncate.
```suggestion
    stamina += roll_remainder( stamina_recovery * turns / 6.0f );
```